### ISSUE TITLE
HMR not working - Full reload needed

### ISSUE THREAD

**Issue by adjnor** [mxstbr/react-boilerplate#1193]
- Node: 7.0.0
- NPM: 3.10.8
- Browser: Chrome 54.0.2840.71 (64-bit)

Changing code (e.g. message text or adding a `<h1>`) in index.js in the Homepage container after a fresh install doesn't do anything.

Steps to reproduce:
1. git clone --depth=1 https://github.com/mxstbr/react-boilerplate.git
2. npm run setup
3. npm start
4. Go to localhost:3000 
5. Change some code -> Nothing happens (Get warning in console: `The following modules couldn't be hot updated: (Full reload needed)`)

I would gladly submit a PR for this but I'm having a hard time figuring out how to fix this

**Comment by gihrig** [comment]
Are you making changes to a functional component? HMR doesn't work there, see #452

**Comment by adjnor** [comment]
No, it's a fresh install, I didn't change anything besides some code in the render method and got the error. Homepage is a PureComponent:

`export class HomePage extends React.PureComponent {`

**Comment by gihrig** [comment]
@adjnor `PureComponent` support was recently added and HMR beakage seems to acknowledged and accepted, see:  this [#811 comment](https://github.com/mxstbr/react-boilerplate/pull/811#issuecomment-238601688).

Could you change that `PureComponent` to a `Component` and let us know what happens?

**Comment by nhducit** [comment]
I have the same problem when I made changed in `action.js`.
react-boilerplate: v3.3.1
Reproduce
```
git clone --depth=1 https://github.com/mxstbr/react-boilerplate.git
npm run setup
npm start
Go to localhost:3000
Change some action code -> Nothing happens (Get warning in console: The following modules couldn't be hot updated: (Full reload needed))
```

**Comment by adjnor** [comment]
@gihrig I tried changing `PureComponent` to `Component` and it did work! I would rather use stateless components though, I think react hot loader 3 is needed for this, saw the issue #1068 , I'll try to look into it. Only thing is changing the defaultMessage text in messages.js doesn't do anything, even on page reload  actually. Changing text in en.json does work but needs a page reload, maybe this is expected behaviour?

**Comment by vincaslt** [comment]
If the text is defined in `en.json` with the same id as in `messages.js`, it will take the one defined in `en.json`. Default should be seen when there is no message defined in `messages.js`.

**Comment by adjnor** [comment]
@vincaslt Alright got it! I'm closing this issue since it seems like the issue is known and a fix is on the way (#1068). For now `Component` will do, thanks!

**Comment by baldore** [comment]
@adjnor I think it's the expected behavior. I've been checking this project (which btw is extremely awesome!!! Thanks!!!) and when I try to change the languages messages, the HMR doesn't work. From my side, the changes that didn't worked (and I was expecting to work) were those I made in RepoListItem.
